cmake/modules/SIMDExt.cmake

```diff
@@ -34,28 +35,44 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
       return ret;
     }
     int main() { foo(0); }" HAVE_ARMV8_CRYPTO)
+
   set(CMAKE_REQUIRED_QUIET ${save_quiet})
   if(HAVE_ARMV8_CRC)
     message(STATUS " aarch64 crc extensions supported")
   endif()
+
   if(HAVE_ARMV8_CRYPTO)
     message(STATUS " aarch64 crypto extensions supported")
   endif()
-  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
+  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH)
+
+  # don't believe only the -march support; gcc 4.8.5 on RHEL/CentOS says
+  # it supports +crc but hasn't got the intrinsics or arm_acle.h.  Test for
+  # the actual presence of one of the intrinsic functions.
+  if(HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH)
+    check_cxx_source_compiles("
+      #include <inttypes.h>
+      int main() { uint32_t a; uint8_t b; __builtin_aarch64_crc32b(a, b); }
+    " HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
+  endif()
```

Comment:
Answer: if you don't pass -march=armv8-a+crc, you get an error from assembly.

`$ cat test.c
#include <inttypes.h>
int main() { uint32_t a; uint8_t b; __builtin_aarch64_crc32b(a, b); }
$ gcc test.c -o test
test.c: In function ‘main’:
test.c:2:1: error: unrecognizable insn:
 int main() { uint32_t a; uint8_t b; __builtin_aarch64_crc32b(a, b); }
 ^
(insn 7 6 8 2 (set (reg:SI 75)
        (unspec:SI [
                (reg:SI 76)
                (reg:QI 77)
            ] UNSPEC_CRC32B)) test.c:2 -1
     (nil))
test.c:2:1: internal compiler error: in extract_insn, at recog.c:2343
$ gcc -march=armv8-a+crc test.c -o test
$